internals/plan/plan.go

```diff
@@ -649,12 +652,12 @@ func CombineLayers(layers ...*Layer) (*Layer, error) {
 				Message: fmt.Sprintf("plan service %q command invalid: %v", name, err),
 			}
 		}
-		if !validServiceAction(service.OnSuccess) {
+		if !(validServiceAction(service.OnSuccess) || service.OnSuccess == ActionFailureShutdown) {
```

Comment:
Not a fan of how this conditional reads. Please consider re-writing the `validServiceAction` func to take a varadic list of additional valid service actions.

```suggestion
		if !validServiceAction(service.OnSuccess, ActionFailureShutdown) {
```